File: src/main/java/Solrj.java
import model.Recipe;
import config.Config;
import org.apache.solr.client.solrj.SolrClient;
import org.apache.solr.client.solrj.SolrServerException;
import org.apache.solr.client.solrj.impl.ConcurrentUpdateSolrClient;
import org.apache.solr.client.solrj.impl.HttpSolrClient;
import org.apache.solr.client.solrj.response.UpdateResponse;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Solrj {

    private static List<Recipe> getRecipe() {
        List<Recipe> ret = new ArrayList<>();
        Recipe r = new Recipe();
        r.id = "hogehoge1";
        r.name = "回鍋肉";
        r.ingredients.add("きゃべつ");
        r.ingredients_amount.add("半分");
        r.steps.add("hogehoge");
        r.genre = "Chinese";
        r.external_info_url = null;
        r.external_info_title = null;
        ret.add(r);

        Recipe curry = new Recipe();
        curry.id = "hogehoge2";
        curry.name = "カレー";
        curry.ingredients.add("たまねぎ");
        curry.ingredients.add("豚肉");
        curry.ingredients_amount.add("半分");
        curry.ingredients_amount.add("200g");
        curry.steps.add("たまねぎを炒める");
        curry.genre = "Japanese";
        curry.external_info_url = "https://www.youtube.com/watch?v=BcKNwo7bzJ4";
        curry.external_info_title = "我が家のカレーがこれになってしまったと何度も言われたほどウマい、市販のルーで作る【至高のカレー】";
        ret.add(curry);

        return ret;
    }

    public static void main(String[] args) throws IOException, SolrServerException {
        Config config = Config.init(System.getenv());

        String solrUrl = String.format("%s/solr/recipe", config.solrServer);
        final ConcurrentUpdateSolrClient solrClient = new ConcurrentUpdateSolrClient.Builder(solrUrl).build();
        try {
            List<Recipe> rs = Solrj.getRecipe();
            UpdateResponse updateResponse = solrClient.addBeans(rs);

            if (updateResponse.getStatus() == 0) {
                updateResponse = solrClient.commit();
                System.out.printf("%s %s", updateResponse.getStatus(), updateResponse.getResponse());
            }
        } catch (SolrServerException e) {
            e.printStackTrace();
            solrClient.rollback();
        } finally {
            solrClient.close();
        }
    }
}

File: src/main/java/Solrj.kt
import config.Config
import model.Genre
import model.Recipe
import kotlin.Throws
import java.io.IOException
import org.apache.solr.client.solrj.SolrServerException
import kotlin.jvm.JvmStatic
import org.apache.solr.client.solrj.impl.ConcurrentUpdateSolrClient
import java.util.ArrayList

object Solrj {
    private fun getRecipe(): List<Recipe> {

        return listOf(
                Recipe(
                        id = "hogehoge1",
                        name = "回鍋肉",
                        Genre = Genre.Chinese,
                        ingredients = listOf("きゃべつ"),
                        ingredients_amount = listOf("半分"),
                        steps = listOf("きゃべつを切る。"),
                ),
                Recipe(
                        id = "hogehoge2",
                        name = "カレー",
                        Genre = Genre.Japanese,
                        ingredients = listOf("たまねぎ", "豚肉"),
                        ingredients_amount = listOf("半分", "200g"),
                        steps = listOf("たまねぎを炒める"),
                        external_info_url = "https://www.youtube.com/watch?v=BcKNwo7bzJ4",
                        external_info_title = "我が家のカレーがこれになってしまったと何度も言われたほどウマい、市販のルーで作る【至高のカレー】"
                )
        )
    }

    @Throws(IOException::class, SolrServerException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val config = Config(System.getenv())
        val solrUrl = String.format("%s/solr/recipe", config.solrServer)
        val solrClient = ConcurrentUpdateSolrClient.Builder(solrUrl).build()
        try {
            var updateResponse = solrClient.addBeans(getRecipe())
            if (updateResponse.status == 0) {
                updateResponse = solrClient.commit()
                System.out.printf("%s %s", updateResponse.status, updateResponse.response)
            }
            solrClient.commit()
        } catch (e: SolrServerException) {
            e.printStackTrace()
            solrClient.rollback()
        } finally {
            solrClient.close()
        }
    }
}